### PR TITLE
MLPAB-704: Certificate provider check your name

### PR DESCRIPTION
**Pull request by acsauk**
# Purpose

Adds page for the CP to check the name the Donor has provided.

Fixes MLPAB-704

## Approach

We're sending an email to the Donor when the CP provides a new name. When we come round to work on this feature properly (how Donor accepts changes from the CP) we'll add some logic around when to send the email based on the extent of the name change.

### CODE REVIEW

File: app/internal/page/certificateprovider/register.go [hunk 1]
```diff
@@ -6,6 +6,8 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/notify"
+
```
Comment (hawx) [inline app/internal/page/certificateprovider/register.go:10]: one day can you figure out why your import adding sometimes puts in blank lines

File: web/template/certificate_provider_check_your_name.gohtml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{{ template "page" . }}
+
```
Comment (hawx) [inline web/template/certificate_provider_check_your_name.gohtml:2]: need to add the `{{ define "serviceName" }}{{ tr .App "beACertificateProvider" }}{{ end }}` block in

File: web/template/certificate_provider_check_your_name.gohtml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    <div class="govuk-grid-row">
+        <div class="govuk-grid-column-two-thirds">
+            <h1 class="govuk-heading-xl govuk-!-margin-bottom-6">{{ tr .App "checkAndConfirmYourName" }}</h1>
+
+            {{ trFormatHtml .App "checkAndConfirmYourNameContent" "DonorFullName" .Lpa.You.FullName "CertificateProviderFullName" .Lpa.CertificateProvider.FullName }}
+
+            <form novalidate method="post">
+                <div class="govuk-form-group {{ if .Errors.Has "is-name-correct" }}govuk-form-group--error{{ end }} govuk-!-margin-bottom-6">
+                    <fieldset class="govuk-fieldset">
+                        {{ template "error-message" (errorMessage . "is-name-correct") }}
```
Comment (hawx) [inline web/template/certificate_provider_check_your_name.gohtml:15]: I was going to say this should go below the legend, but looks like the CSS makes it happen anyway TIL

**Review by hawx — COMMENTED** [review]
shouldn't this page be in the flow?

File: app/internal/page/certificateprovider/register.go (hunk 1 above)
Comment (acsauk) [reply app/internal/page/certificateprovider/register.go:10]: Hahaha I'll ask JetBrains

File: web/template/certificate_provider_check_your_name.gohtml (hunk 3 above)
Comment (acsauk) [reply web/template/certificate_provider_check_your_name.gohtml:15]: Yeah I was super confused for a second there until I saw what was going on.

File: web/template/certificate_provider_check_your_name.gohtml (hunk 2 above)
Comment (acsauk) [reply web/template/certificate_provider_check_your_name.gohtml:2]: I'd totally missed we were even doing this - added in now.

**Review by hawx — APPROVED** [review]
although, should this be the name used on the "your details" page you go to next?

**Pull request merged**